src/util.rs

```diff
@@ -51,14 +52,22 @@ fn expect_property(obj: &mut JsonObject, name: &'static str) -> Result<JsonValue
 fn expect_owned_array(value: JsonValue) -> Result<Vec<JsonValue>, Error> {
     match value {
         JsonValue::Array(v) => Ok(v),
-        _ => Err(Error::ExpectedArrayValue),
+        _ => match value {
+            // it can never be Array, but that's exhaustive matches for you
```

Comment:
😂

Reply:
- [x] minimizes the number of distinct error types
 - [x] produces useful error messages
 - [x] avoids clone

🎉